### DISCUSSION TITLE
Can't sign and a file with unknown attributes

### DISCUSSION THREAD

**Discussion by GregStevenson** [mkaring/ConfuserEx#441] [Q&A]
Have a project with about 50 assemblies. After obfuscating, they can all be signed except four. Normally we just tweak the crproj file adding lines like:  <module path="YetAnotherModule.dll" /> However, today I opened up the UI and noticed something on the left side of the project tab. All the modules have detail info (apparently extracted from the assembly itself) except for our four problem DLLs. Instead of detail information they have "Unknown". I checked these DLLs and they all have assembly info and it shows in the details tab of a properties dialog. 

1) What could be the reason for the "Unknown" and how to correct?
2) Why would this state prevent signtool from being able to sign it after obfuscation. The un-obfuscated files can be signed without issue.

**Comment by mkaring** [comment]
Unknown means that the assembly can't be read as an .NET assembly. This means usually that the header part of the .NET assembly is missing.
If the signtool doesn't work his point to the file not being an DLL at all. It's really hard to say, what is going on there. You say before the obfuscation the assemblies are fine and they are broken only after the obfuscation?
Is there maybe an anti virus application that is messing with the files while ConfuserEx is working or after that?

Is it possible for you to send me those files for analysis?

**Reply by GregStevenson** [reply]
Thanks for the reply. No AV software that would cause issues. I do have it scheduled to compile and add debugging to the ConfuserEx code this week. Then I'll try to find our what these files have in common. I'll report back shortly after Christmas.
